portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,30 +19,91 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.List;
 import java.util.ArrayList;
+import java.util.Date;
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private String[] retrieveComments() {
-        ArrayList<String> comments = new ArrayList<String>();
-        comments.add("user1: hey what's up");
-        comments.add("user2: no way, you're joking!");
-        comments.add("user3: wow, great job.");
-        return comments.toArray(new String[comments.size()]);
-    } 
+    private List<Comment> comments;
+    private List<Long> ids;
+
+    public DataServlet() {
+        comments = new ArrayList<Comment>();
+        ids = new ArrayList<Long>();
+    }
```

Comment:
Keeping state of the comments and ids in your DataServlet will work for one user, but I worry that it won't work as expected for multiple users. If you open multiple tabs or browsers and try to view the comments at the same time, can both tabs see all the comments?

Reply:
Ah this was just to fix a bug that was posting the same comment twice. Otherwise, it has no other purpose and isn't meant to emulate simultaneous commenters.

Reply:
Since WebServlets are intended to be used by multiple users simultaneously, give it a try at making your DataServlet work for more than one open tab.